Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/usr/local/lib/python3.9/site-packages/nbclient/client.py", line 1093, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/usr/local/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/usr/local/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/usr/local/Cellar/python@3.9/3.9.18/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.9/site-packages/nbclient/client.py", line 559, in async_execute
    await self.async_execute_cell(
  File "/usr/local/lib/python3.9/site-packages/nbclient/client.py", line 854, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/usr/local/lib/python3.9/site-packages/nbclient/client.py", line 756, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import numpy as np
from scipy.stats import multivariate_normal
import matplotlib.pyplot as plt
import matplotlib

X = np.genfromtxt('mixture.csv', delimiter=',')
K = 3
n, d = X.shape

# Param√®tres initiaux
W = np.array([[1/3,1/3,1/3]]) # poids
M  = np.array([[-2.0,-4,0],[-3,1,-1]]) # Moyennes
C = np.zeros((3,2,2)) # Co

C[:,0,0] = 1
C[:,1,1] = 1

p = np.zeros((3,300))

for i in range(0,100): 

    # Etape E
    for k in range(0,K):  
        mvn = multivariate_normal( M[:,k].T, C[k,:,:] )
        p[k,:] = W[0,k]*mvn.pdf(X)

    # Etape M
    p = p/sum(p,0)   
    W = np.mean(p,1).reshape(1,3)
    for k in range(0,K):
        M[:,k] = (X.T @ p[k,:].T)/sum(p[k,:])
        xm = X.T - M[:,k].reshape(2,1)
        C[k,:,:] = xm @ (xm*p[k,:]).T/sum(p[k,:])


fig = plt.subplot(1, 1, 1)
plt.scatter(X[:,0],X[:,1])
c = ['r','b','g']
for i in range(0,3):
    v, w = np.linalg.eigh(C[i,:,:])
    v = 2.0 * np.sqrt(2.0) * np.sqrt(v)
    u = w[0] / np.linalg.norm(w[0])

    angle = np.arctan(u[1] / u[0])
    angle = 180.0 * angle / np.pi  
    ell = matplotlib.patches.Ellipse(M[:,i], v[0], v[1], angle=180.0 + angle, color=c[i])
    ell.set_clip_box(fig.bbox)
    ell.set_alpha(0.5)
    fig.add_artist(ell)

plt.tight_layout()
plt.show()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mOSError[0m                                   Traceback (most recent call last)
Input [0;32mIn [7][0m, in [0;36m<module>[0;34m[0m
[1;32m      3[0m [38;5;28;01mimport[39;00m [38;5;21;01mmatplotlib[39;00m[38;5;21;01m.[39;00m[38;5;21;01mpyplot[39;00m [38;5;28;01mas[39;00m [38;5;21;01mplt[39;00m
[1;32m      4[0m [38;5;28;01mimport[39;00m [38;5;21;01mmatplotlib[39;00m
[0;32m----> 6[0m X [38;5;241m=[39m [43mnp[49m[38;5;241;43m.[39;49m[43mgenfromtxt[49m[43m([49m[38;5;124;43m'[39;49m[38;5;124;43mmixture.csv[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[43mdelimiter[49m[38;5;241;43m=[39;49m[38;5;124;43m'[39;49m[38;5;124;43m,[39;49m[38;5;124;43m'[39;49m[43m)[49m
[1;32m      7[0m K [38;5;241m=[39m [38;5;241m3[39m
[1;32m      8[0m n, d [38;5;241m=[39m X[38;5;241m.[39mshape

File [0;32m/usr/local/lib/python3.9/site-packages/numpy/lib/npyio.py:1793[0m, in [0;36mgenfromtxt[0;34m(fname, dtype, comments, delimiter, skip_header, skip_footer, converters, missing_values, filling_values, usecols, names, excludelist, deletechars, replace_space, autostrip, case_sensitive, defaultfmt, unpack, usemask, loose, invalid_raise, max_rows, encoding, like)[0m
[1;32m   1791[0m     fname [38;5;241m=[39m os_fspath(fname)
[1;32m   1792[0m [38;5;28;01mif[39;00m [38;5;28misinstance[39m(fname, [38;5;28mstr[39m):
[0;32m-> 1793[0m     fid [38;5;241m=[39m [43mnp[49m[38;5;241;43m.[39;49m[43mlib[49m[38;5;241;43m.[39;49m[43m_datasource[49m[38;5;241;43m.[39;49m[43mopen[49m[43m([49m[43mfname[49m[43m,[49m[43m [49m[38;5;124;43m'[39;49m[38;5;124;43mrt[39;49m[38;5;124;43m'[39;49m[43m,[49m[43m [49m[43mencoding[49m[38;5;241;43m=[39;49m[43mencoding[49m[43m)[49m
[1;32m   1794[0m     fid_ctx [38;5;241m=[39m contextlib[38;5;241m.[39mclosing(fid)
[1;32m   1795[0m [38;5;28;01melse[39;00m:

File [0;32m/usr/local/lib/python3.9/site-packages/numpy/lib/_datasource.py:193[0m, in [0;36mopen[0;34m(path, mode, destpath, encoding, newline)[0m
[1;32m    156[0m [38;5;124;03m"""[39;00m
[1;32m    157[0m [38;5;124;03mOpen `path` with `mode` and return the file object.[39;00m
[1;32m    158[0m 
[0;32m   (...)[0m
[1;32m    189[0m 
[1;32m    190[0m [38;5;124;03m"""[39;00m
[1;32m    192[0m ds [38;5;241m=[39m DataSource(destpath)
[0;32m--> 193[0m [38;5;28;01mreturn[39;00m [43mds[49m[38;5;241;43m.[39;49m[43mopen[49m[43m([49m[43mpath[49m[43m,[49m[43m [49m[43mmode[49m[43m,[49m[43m [49m[43mencoding[49m[38;5;241;43m=[39;49m[43mencoding[49m[43m,[49m[43m [49m[43mnewline[49m[38;5;241;43m=[39;49m[43mnewline[49m[43m)[49m

File [0;32m/usr/local/lib/python3.9/site-packages/numpy/lib/_datasource.py:533[0m, in [0;36mDataSource.open[0;34m(self, path, mode, encoding, newline)[0m
[1;32m    530[0m     [38;5;28;01mreturn[39;00m _file_openers[ext](found, mode[38;5;241m=[39mmode,
[1;32m    531[0m                               encoding[38;5;241m=[39mencoding, newline[38;5;241m=[39mnewline)
[1;32m    532[0m [38;5;28;01melse[39;00m:
[0;32m--> 533[0m     [38;5;28;01mraise[39;00m [38;5;167;01mIOError[39;00m([38;5;124m"[39m[38;5;132;01m%s[39;00m[38;5;124m not found.[39m[38;5;124m"[39m [38;5;241m%[39m path)

[0;31mOSError[0m: mixture.csv not found.
OSError: mixture.csv not found.

